plugins/calibre.koplugin/metadata.lua

```diff
@@ -242,6 +242,7 @@ function CalibreMetadata:init(dir, is_search)
 
     local msg
     if is_search then
+        self:cleanUnused()
```

Comment:
mmm, that removes entries from the json file, not from the cache itself. The cache is always slimed and `cleanUnused` prunes unused entries from the table before flushing the table to the json file. See https://github.com/koreader/koreader/blob/master/plugins/calibre.koplugin/metadata.lua#L207

What is the problem with current implementation?.

Reply:
`removeNull` only removes NULLs from `series` & `series_index`; if we don't do this first, we have many more fields with potential NULLs.

(This is what happened before #7159. Plus, we don't actually need those extra fields to be dumped to the cache, and `cleanUnused` only cleans the internal representiation of the loaded JSON, not the original file).

Reply:
Oh, wait, yes, it does dump back to the file. >_<".

Reply:
(Well, no, it didn't actually go through cleanUnused before #7159, but, still ;p).

Reply:
Instead, `getAllMetadata` duplicated cleanUnused'd behavior, without the dump, so, this is pretty much the same thing, only DRYer ;).